chain/network/src/routing/graph_with_cache.rs

```diff
@@ -115,4 +116,21 @@ impl GraphWithCache {
         }
         edges
     }
+
+    pub fn prune_old_edges(&mut self, prune_edges_older_than: Utc) {
+        let old_edges = self
+            .edges
+            .iter()
+            .filter_map(|(edge_key, edge)| {
+                if edge.is_edge_older_than(prune_edges_older_than) {
+                    Some(edge_key.clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
not sure what is the impact, but calling repeatedly remove_edge reacquires a mutex each time.

Reply:
fixed.